app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -138,7 +138,14 @@ class AuditController @Inject() (
       (for {
         string  <- extractBody(request.body)
         details <- parse[Details](string)
-        result  <- auditService.sendStatusTypeEvent(details, auditType.name, auditSource).asPresentation
+        parentAuditType = if (auditType.parent.isDefined) auditType.parent.get else "None"
+        result <- auditService
+          .sendStatusTypeEvent(
+            details.copy(details.metadata.copy(subtype = Some(auditType.name))),
+            parentAuditType,
+            auditSource
+          )
+          .asPresentation
```

Comment:
No -- what you've done here is make every status type that _doesn't_ have a parent defined have a parent of `None` -- which isn't right at all, plus line 141 could have simply been written `auditType.parent.getOrElse("None")`... but please don't do that, because that's still wrong.

Really, what you want to do is:

* Determine if there is a parent type -- if so, that's the audit type, else `auditType.name` is (so `auditType.parent.getOrElse(auditType.name)`.
* If there is an audit type, morph the details object. As `subtype` will never be provided by the consumer, it should not be on the request object. Currently, it's vulnerable to "overposting", as the `subType` is calculated within auditing and should _never_ be specified by the user. 
* With that mind, I suggest creating a `DetailsRequest` object for handing incoming requests from API/router, and make the `Details` object from the `DetailsRequest` that does the calculation for you.